src/test/java/org/dependencytrack/resources/v1/AnalysisResourceTest.java

```diff
@@ -471,16 +497,14 @@ public void updateAnalysisCreateNewWithEmptyRequestTest() throws Exception {
         assertThat(responseJson.getJsonArray("analysisComments")).isEmpty();
         assertThat(responseJson.getBoolean("isSuppressed")).isFalse();
 
-        assertConditionWithTimeout(() -> NOTIFICATIONS.size() == 2, Duration.ofSeconds(5));
+        assertConditionWithTimeout(() -> NOTIFICATIONS.size() == 1, Duration.ofSeconds(5));
         final Notification projectNotification = NOTIFICATIONS.poll();
         assertThat(projectNotification).isNotNull();
-        final Notification notification = NOTIFICATIONS.poll();
-        assertThat(notification).isNotNull();
-        assertThat(notification.getScope()).isEqualTo(NotificationScope.PORTFOLIO.name());
-        assertThat(notification.getGroup()).isEqualTo(NotificationGroup.PROJECT_AUDIT_CHANGE.name());
-        assertThat(notification.getLevel()).isEqualTo(NotificationLevel.INFORMATIONAL);
```

Comment:
No `PROJECT_AUDIT_CHANGE` notification being emitted here seems like a breaking change. Was its emission a bug that is now resolved?

Reply:
That's a valid concern, I will investigate.

Reply:
Okay, in this particular case, the submitted analysis has no fields set (the test name even says this, `updateAnalysisCreateNewWithEmptyRequestTest`). With the new code, `null` and `NOT_SET` are handled slightly different. Semantically, I don't think this should lead to a `PROJECT_AUDIT_CHANGE` if nothing actually changed.

Reply:
Thanks for investigating, makes sense.